web-service/helpers.py

```diff
@@ -380,6 +385,18 @@ def StoreUrl(siteInfo, url, content, encoding):
 
 ################################################################################
 
+def FaviconUrl(url):
+    # Make sure we only fetch favicons we've already added to our database.
```

Comment:
This check is only made because the `favicon_url` is added into the database only if it's a valid favicon image based on the only facts its is indeed a publicly accessible URL that returns a blob whose contentType starts with `image/`.